transmogrifier/sources/xml/dspace_dim.py

```diff
@@ -260,47 +183,174 @@ def get_optional_fields(self, xml: Tag) -> dict | None:
         return fields
 
     @classmethod
-    def get_content_types(cls, xml: Tag) -> list[str] | None:
-        """
-        Retrieve content types from a DSpace DIM XML record.
+    def get_alternate_titles(
+        cls, source_record: Tag
+    ) -> list[timdex.AlternateTitle] | None:
+        alternate_titles = [
+            timdex.AlternateTitle(
+                value=str(alternate_title.string),
+                kind=alternate_title["qualifier"],
+            )
+            for alternate_title in source_record.find_all(
+                "dim:field", element="title", string=True
+            )
+            if alternate_title.get("qualifier")
+        ]
+        # If the record has more than one main title, add extras to alternate_titles
+        alternate_titles.extend(
+            [
+                timdex.AlternateTitle(value=title)
+                for title in cls.get_main_titles(source_record)[1:]
+            ]
+        )
+        return alternate_titles or None
 
-        May be overridden by source subclasses that retrieve content type values
-        differently.
+    @classmethod
+    def get_citation(cls, source_record: Tag) -> str | None:
+        if citation := source_record.find(
+            "dim:field", element="identifier", qualifier="citation", string=True
+        ):
+            return citation.string
+        return None
 
-        Args:
-            xml: A BeautifulSoup Tag representing a single DSpace DIM XML record.
-        """
+    @classmethod
+    def get_content_type(cls, source_record: Tag) -> list[str] | None:
         return [
-            t.string for t in xml.find_all("dim:field", element="type", string=True)
+            str(content_type.string)
+            for content_type in source_record.find_all(
+                "dim:field", element="type", string=True
+            )
         ] or None
 
     @classmethod
-    def get_main_titles(cls, xml: Tag) -> list[str]:
+    def get_contents(cls, source_record: Tag) -> list[str] | None:
+        return [
+            contents.string
+            for contents in source_record.find_all(
+                "dim:field",
+                element="description",
+                qualifier="tableofcontents",
+                string=True,
+            )
+        ] or None
+
+    @classmethod
+    def get_contributors(cls, source_record: Tag) -> list[timdex.Contributor] | None:
+        contributors = []
+        contributors.extend(list(cls._get_creators(source_record)))
+        contributors.extend(
+            list(cls._get_contributors_by_contributor_element(source_record))
+        )
+        return contributors or None
+
+    @classmethod
+    def _get_creators(cls, source_record: Tag) -> Iterator[timdex.Contributor]:
+        for creator in source_record.find_all(
+            "dim:field", element="creator", string=True
+        ):
+            yield timdex.Contributor(
+                value=str(creator.string),
+                kind="Creator",
+            )
```

Comment:
Just curious (and same for `_get_contributors...()` below), why the `yield`?

On one hand, I kind of like it: no need for list comprehensions, just yield `Contributor` objects as they are found.  On the other hand: feels a bit different than other places?

FWIW, I do kind of like the `yield`?  Wonder -- not pressing by any means -- if opportunity to move in this direction for other methods as we encounter them?

Reply:
Yeah, those might make for a couple of nice PRs when the first pass is over: "Normalize yield..",  "Normalize BS4 string casting...", etc.  Though I think by and large things are pretty good.